sourcegraph/cody
# Edit: Strip HTML entities before applying response

## Description

Context: https://sourcegraph.slack.com/archives/C06R69BC8UW/p1715101999120499

Edits can sometimes include HTML entities in the inserted text. This appears to be more prevalent in certain languages, and is due to the LLM responding with Markdown.

We strip the Markdown tags, but we do not decode the text.

## Test plan

Added unit tests

1. Make Edits on a range of code that uses `<` or `>`. If needed, encourage the LLM to respond with Markdown. Ideally we return the correct code regardless of if we receive markdown or not.
2. Check the inserted code is as expected
